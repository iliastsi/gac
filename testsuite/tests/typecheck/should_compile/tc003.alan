unreachable_code() : proc
  foo1() : proc
  {
    return;
    writeString("foo\n");
  }

  foo2() : proc
  {
    if(true)
      return;
    else
      return;
    writeString("foo2\n"); }

  foo3() : proc
  {
    if(true)
      return;
    else
      writeString("false\n");
    writeString("foo3\n");
  }

  foo4() : proc
  {
    while(true)
      return;
    writeString("foo4\n");
  }
{
  foo1();
  foo2();
  foo3();
  foo4();
  return;

  writeString("False\n");
  writeString("unreachable_code\n");
}
