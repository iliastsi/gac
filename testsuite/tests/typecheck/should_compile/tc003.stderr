tc003.alan:5:5-25: Warning: Unreachable code
	Dead code has been eliminated

tc003.alan:14:5-26: Warning: Unreachable code
	Dead code has been eliminated

tc003.alan:38,3-39,36: Warning: Unreachable code
	Dead code has been eliminated

